lib/dotcom/system_status/groups.ex

```diff
@@ -0,0 +1,465 @@
+  end
+
+  # - If the active period is in the future, returns its start_time.
+  # - If the active period indicates that the alert is currently active, returns nil.
+  # - Raises an error if the alert is completely in the past.
+  defp future_start_time(
+         [{start_time, _end_time} = first_active_period | more_active_periods],
+         time
+       ) do
+    cond do
+      ends_before?(first_active_period, time) -> future_start_time(more_active_periods, time)
+      starts_before?(first_active_period, time) -> nil
+      true -> start_time
+    end
+  end
+
+  # Returns true if the active period ends before the time given. An
+  # end-time of false indicates an indefinite active period, which
+  # never ends.
+  defp ends_before?({_start_time, nil}, _time), do: false
+  defp ends_before?({_start_time, end_time}, time), do: Timex.before?(end_time, time)
+
+  # Returns true if the active period starts before the time given.
+  defp starts_before?({start_time, _end_time}, time), do: Timex.before?(start_time, time)
+
+  # Combines statuses that have the same active time and description
+  # into a single pluralized status (e.g. "Station Closures" instead
+  # of "Station Closure").
+  defp consolidate_duplicate_descriptions(statuses) do
+    statuses
+    |> Enum.group_by(fn %{time: time, description: description} -> {time, description} end)
+    |> Enum.map(fn
+      {_, [status]} -> status
+      {_, [status | _]} -> pluralize_description(status)
+    end)
+  end
+
+  # Replaces the description of a status with its plural form, for use
+  # in &consolidate_duplicate_descriptions/2, when multiple statuses
+  # have the same time and effect.
+  defp pluralize_description(%{description: description} = status) do
+    %{status | description: Inflex.pluralize(description)}
+  end
+
+  # Sorts the given list of statuses first by time, then by
+  # description, so that earlier statuses show up before later ones,
+  # and then to keep statuses in a stable order.
+  #
+  # This takes advantage of the fact that `nil` is sorted before
+  # anything else, which allows it to automatically sort active
+  # statuses before future ones.
+  # 
+  # This should be called before &stringify_times/1, otherwise times
+  # will get sorted lexically instead of temporally (e.g. 10:00pm will
+  # get sorted ahead of 9:00pm).
+  defp sort_statuses(statuses) do
+    statuses
+    |> Enum.sort_by(fn %{time: time, description: description} -> {time, description} end)
+  end
+
+  # Transforms the time in each status given into a human-readable
+  # string or nil.
+  defp stringify_times(statuses) do
+    statuses
+    |> Enum.map(fn status ->
+      %{
+        status
+        | time: stringify_time(status.time)
+      }
+    end)
+  end
+
+  # Returns a human-readable version of the time given, formatted like
+  # "8:30pm", for example. Leaves nil unchanged.
+  defp stringify_time(nil), do: nil
+  defp stringify_time(time), do: Util.kitchen_downcase_time(time)
```

Comment:
I don't like the fact that stringify_time takes something that isn't a time and returns something that isn't a string.

Reply:
Do you have a better name in mind? `maybe_time_to_prefix` (assuming [this suggestion](https://github.com/mbta/dotcom/pull/2330#discussion_r1931390295))?

Reply:
Talked offline and agreed to move all of the presentational stringifying logic into widget/frontend layer, so this is gone now.